number_parser/parser.py

```diff
@@ -81,16 +115,18 @@ def _build_number(token_list, lang_data):
             current_grp_value += lang_data.hundreds[token]
 
         elif token in lang_data.big_powers_of_ten:
+            power_of_ten = lang_data.big_powers_of_ten[token]
             if current_grp_value == 0:
                 current_grp_value = 1
 
-            current_grp_value *= lang_data.big_powers_of_ten[token]
-            if not lang_data.big_powers_of_ten[token] == 100:
+            current_grp_value *= power_of_ten
+            if power_of_ten > 100:
```

Comment:
This is basically group value being decided in case in English we see a power of 10 greater than 100. (i.e thousand , million , billion etc) we break. However for spanish this needs to ``` if power_of_ten > 100000``` as thousand can be used with a larger power later on.